syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.apache.skywalking.apm.network.proto";


//register service for ApplicationCode, this service is called when service starts.
service ApplicationRegisterService {
    rpc applicationCodeRegister (Application) returns (ApplicationMapping) {
    }
}

message KeyWithIntegerValue {
    string key = 1;
    int32 value = 2;
}

message Application {
    string applicationCode = 1;
}

message ApplicationMapping {
    KeyWithIntegerValue application = 1;
}
